src/steps/Microsoft.DotNet.UpgradeAssistant.Steps.Packages/Analyzers/NewtonsoftReferenceAnalyzer.cs

```diff
@@ -36,21 +36,28 @@ public async Task<PackageAnalysisState> AnalyzeAsync(IProject project, PackageAn
                 throw new ArgumentNullException(nameof(state));
             }
 
-            if (!project.Components.HasFlag(ProjectComponents.AspNetCore))
+            // This reference only needs added to ASP.NET Core exes
+            if (!project.Components.HasFlag(ProjectComponents.AspNetCore) || project.OutputType != ProjectOutputType.Exe)
```

Comment:
nit: all the negation here feels hard to weird....

Reply:
I'll De Morgan it to have less !.